compiler/rustc_codegen_llvm/src/builder.rs

```diff
@@ -898,6 +898,30 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         unsafe { llvm::LLVMBuildTrunc(self.llbuilder, val, dest_ty, UNNAMED) }
     }
 
+    fn unchecked_utrunc(&mut self, val: &'ll Value, dest_ty: &'ll Type) -> &'ll Value {
+        let trunc = self.trunc(val, dest_ty);
+        if llvm_util::get_version() >= (19, 0, 0) {
+            unsafe {
+                if llvm::LLVMIsATruncInst(trunc).is_some() {
```

Comment:
Checking LLVMIsAInstruction would be fine as well, don't really need to export the extra API.

Reply:
Okay, that probably means that you have some cases with a trunc to the same type. In that case, you will end up trying to set the flag on an unrelated instruction (producing a crash with LLVMIsAInstruction -- with LLVMIsATruncInst it might end up setting the flag on an unrelated trunc).

You'll want to check for the no-op trunc case and bail out early.